### DISCUSSION TITLE
Implement a importable Sanitize (Slugify) Function

### DISCUSSION THREAD

**Discussion by nico-k3** [gatsbyjs/gatsby#32926] [Ideas / Feature Requests]
Hi Guys!

Sometimes it can be tricky to source data from a Backend and create pages with let's say the title. Content creators are usually not familiar with things that can go wrong and tend to use some patterns that give headaches to us. They use emojis, brackets,  slashes, etc.

Fortunately, there are some helpers like [slugify ](https://github.com/simov/slugify) or [@sindresorhus/slugify](https://github.com/sindresorhus/slugify) to get rid of the nasty stuff and create clean versions of the input String. Unfortunately, though they are not made for Gatsby and Gatsby itself has some special behaviors.

As an example: when today one of the editors published an article with a title like "This // is nice", Gatsby interpreted it as a subpage and therefore created the page "/articles/this/is-nice" - certainly not a big deal but my slugify solution just erased the slashes from the title and therefore a dead link was published to the page.

Also of course if the Gatsby Team ever decides to change something about their algorithm we all would have to adjust. As Gatsby itself seems to have a solution it uses to sanitize Strings and make them URL ready - wouldn't it be nice if we could just use the same algorithm?

How would you like this feature? A simple importable slugify function inside of the Gatsby core to get rid of the headaches.

Nico

**Comment by iffjey** [comment]
Having the same issue mentioned here https://github.com/gatsbyjs/gatsby/discussions/32820
But for some reason being ignored for almost two weeks 🤷‍♂️

**Reply by nico-k3** [reply]
Not the same, but Related: You would like to set your function as the slug creation algorithm for Gatsby, I would like to use the existing Function Gatsby uses and import it into my projects.

Although If I could set the Function, this would resolve my problem of not knowing what exactly Gatsby does too.

**Reply by KyleAMathews** [reply]
👍 a stricter slugify function sounds great! This isn't entirely a Gatsby concern as any framework that creates URL from data would have the same issue. Maybe start a new project with stricter rules? It's definitely something we could consider baking into Gatsby at some point.

**Reply by iffjey** [reply]
Baked into solutions are great. But as you said «any framework would have the same issues» that's why we need a possibility to provide custom solution. It could be just a function in configuration which is replacing slugify.

In my case, it's a blocker to make a migration from the old engine to Gatsby. Because generated slugs is not the same as the used to be.
